python/src/lib.rs

```diff
@@ -1127,6 +1128,71 @@ impl From<&PyAddAction> for Add {
     }
 }
 
+#[pyfunction]
+#[allow(clippy::too_many_arguments)]
+fn write_to_deltalake(
+    table_uri: String,
+    data: PyArrowType<ArrowArrayStreamReader>,
+    // schema: Option<PyArrowType<ArrowSchema>>, // maybe do the schema casting on python side
+    mode: String,
+    max_rows_per_group: i64,
+    overwrite_schema: bool,
+    partition_by: Option<Vec<String>>,
+    name: Option<String>,
+    description: Option<String>,
+    configuration: Option<HashMap<String, Option<String>>>,
+    storage_options: Option<HashMap<String, String>>,
+) -> PyResult<()> {
+    let batches = data.0.map(|batch| batch.unwrap()).collect::<Vec<_>>();
```

Comment:
We should probably try to not fully materialize the RBR into memory. This would entail wrapping the RBR in an `ExecutionPlan`, which can be passed to the `CreateBuilder`. Maybe [`DatasetExec`](https://github.com/apache/arrow-datafusion-python/blob/main/src/dataset_exec.rs) from datafusion-python can provide some inspiration.

Reply:
Maybe we can do that in a second PR then, since I was also doing it like this for MERGE python binding. Then I can improve these two at the same time.